pkg/controller/controller_utils.go

```diff
@@ -1028,14 +1028,18 @@ func ComputeHash(template *v1.PodTemplateSpec, collisionCount *int32) string {
 	podTemplateSpecHasher := fnv.New32a()
 	hashutil.DeepHashObject(podTemplateSpecHasher, *template)
 
+	sum32 := podTemplateSpecHasher.Sum32()
+
 	// Add collisionCount in the hash if it exists.
 	if collisionCount != nil {
 		collisionCountBytes := make([]byte, 8)
 		binary.LittleEndian.PutUint32(collisionCountBytes, uint32(*collisionCount))
-		podTemplateSpecHasher.Write(collisionCountBytes)
+		if _, err := podTemplateSpecHasher.Write(collisionCountBytes); err == nil {
+			sum32 = podTemplateSpecHasher.Sum32()
+		}
 	}
 
-	return rand.SafeEncodeString(fmt.Sprint(podTemplateSpecHasher.Sum32()))
+	return rand.SafeEncodeString(fmt.Sprint(sum32))
```

Comment:
While I think there was a problem you identified I'm not sure this is correct either. This means if there is a collision and we hit an error trying to write the count used to avoid hash collision, we will return the hash as if there were no collision. That seems more problematic. Given that we are writing to uint32 chunk of memory, do we understand why we are hitting an error?

Reply:
I am still trying to find the implementation of the Write() call so that we can see why there could be an error.
Looking at DeepHashObject(), it ignores the error return from printer.Fprintf.
I think if there is error, it should happen to both Fprintf and Write.

ComputeHash currently doesn't return error. I think the proper fix is to surface potential error from both Fprintf and Write to the caller of ComputeHash.

@cheftako 
I want to get your opinion before making this change.

Reply:
I believe the implementation of this write method can be found at https://golang.org/src/hash/fnv/fnv.go (Line 109?).
I do not think that it will ever return an error.
However if the write method were changed to return an error, the important thing for us is that the hash function NOT return the same value for pod template as it does for the pod template it collides with.
I wouldn't bother to change it but if we wanted to be paranoid we might change it to panic (with a detailed explanation of why this should never happen) if an error were thrown.

Reply:
The Write func's around line 109 in fnv.go return nil as error.
The current code seems to be fine.

Thanks for your time, @cheftako